collector/receiver/cgoreceiver/cgoreceiver.go

```diff
@@ -0,0 +1,204 @@
+package cgoreceiver
+
+/*
+#cgo LDFLAGS: -L ./ -lkindling  -lstdc++ -ldl
+#cgo CFLAGS: -I .
+#include <stdlib.h>
+#include <stdint.h>
+#include <stdio.h>
+#include "cgo_func.h"
+*/
+import "C"
+import (
+	"sync"
+	"time"
+	"unsafe"
+
+	analyzerpackage "github.com/Kindling-project/kindling/collector/analyzer"
+	"github.com/Kindling-project/kindling/collector/component"
+	"github.com/Kindling-project/kindling/collector/model"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/Kindling-project/kindling/collector/receiver"
+)
+
+const (
+	Cgo = "cgoreceiver"
+)
+
+type CKindlingEventForGo C.struct_kindling_event_t_for_go
+
+type CgoReceiver struct {
+	cfg             *Config
+	analyzerManager *analyzerpackage.Manager
+	shutdownWG      sync.WaitGroup
+	telemetry       *component.TelemetryTools
+	eventChannel    chan *model.KindlingEvent
+	eventCount      int
+	stopCh          chan interface{}
+}
+
+type Config struct {
+	SubcribeInfo []SubEvent `mapstructure:"subscribe"`
+}
+
+type SubEvent struct {
+	Category string `mapstructure:"category"`
+	Name     string `mapstructure:"name"`
+}
+
+func NewCgoReceiver(config interface{}, telemetry *component.TelemetryTools, analyzerManager *analyzerpackage.Manager) receiver.Receiver {
+	cfg, ok := config.(*Config)
+	if !ok {
+		telemetry.Logger.Sugar().Panicf("Cannot convert [%s] config", Cgo)
+	}
+	cgoReceiver := &CgoReceiver{
+		cfg:             cfg,
+		analyzerManager: analyzerManager,
+		telemetry:       telemetry,
+		eventChannel:    make(chan *model.KindlingEvent, 3e5),
+		stopCh:          make(chan interface{}, 1),
+	}
+	return cgoReceiver
+}
+
+func (r *CgoReceiver) Start() error {
+	r.telemetry.Logger.Info("Start CgoReceiver")
+	C.runForGo()
+	go r.printMetrics()
+	time.Sleep(2 * time.Second)
+	r.subEvent()
+	// Wait for the C routine running
+	time.Sleep(2 * time.Second)
+	go r.consumeEvents()
+	go r.startGetEvent()
+	return nil
+}
+
+// TODO finish it using opentelemetry
+func (r *CgoReceiver) printMetrics() {
+	timer := time.NewTicker(1 * time.Second)
+	r.shutdownWG.Add(1)
+	for {
+		select {
+		case <-r.stopCh:
+			r.shutdownWG.Done()
+			return
+		case <-timer.C:
+			r.telemetry.Logger.Info("Total number events received: ", zap.Int("events", r.eventCount))
+			r.eventCount = 0
+			r.telemetry.Logger.Info("Current channel size: ", zap.Int("channel size", len(r.eventChannel)))
+		}
+	}
```

Comment:
Was that necessary to print this Metric every second

Reply:
No, I will open another PR to introduce opentelemetry to record metrics.